Макеев Петр, M3337

Good-Morning opened the pull request:
Пока только первое подзадание (TypeScript + CSS Modules).

Commit:
+task5 +TypeScript +CSS_Modules

Good-Morning commented:
**now-cli**: 

> TypeScript error: Could not find a declaration file for module 'react'.

Не уверен, что могу что-то с этим сделать с моей стороны.

Commit:
attempts to fix issue with now-cli

Commit:
+dark-theme

Commit:
+smartSearch

Was extended for searching in letters' text.
Now works without lags for mail sets of more then 1e5 letters.

Commit:
incremental-search-fix aliquot-letter

Fixed: the bug with iterative writing down the result of a searching.
Now every ~10000th letter is a letter with spec marker 'Кратное письмо'.

Commit:
fixes some spec cases of searching

Adding to the filtered letters new ones.

Good-Morning commented:
**1. Сделана темная тема (с сохранением)**

**2. Улучшен поиск:** 
   - добавлен поиск по содержимому
   - процесс поиска разделен в части (setTimeout как yield)
   - поиск начинается, когда было 300ms простоя поля ввода

 **NB:** 
- сейчас **некорректно работает выделение**(и кое-что с этим связанное) - выделяются одинаковые. Это связано не с логикой, а с построением изначального множества писем. Чтобы заметить работу поиска, нужно хотя бы 1e5 элементов . Чтобы это сильно не нагружало машину (создание + хранение), список ссылается на одни и те же объекты.
- в списке писем есть 2 контрольных вида. Их можно найти, введя префиксы строк "**кратное письмо**" и "**последнее письмо**" в поиск. Нужны для демонстрации работы поиска.

KokorinIlya commented:
Тёмная тема

KokorinIlya commented:
Или куда там надо написать, чтобы она появилась? Мне уведомление от сайта не приходит

Good-Morning commented:
> Или куда там надо написать, чтобы она появилась? Мне уведомление от сайта не приходит

![image](https://user-images.githubusercontent.com/24812415/56983971-a0921600-6b8d-11e9-8a2a-c138a1c497a2.png)

Как и сказано в задании:

> Переключатель должен находиться в шапке

Хотя я мог это неправильно интерпретировать.

KokorinIlya commented:
Вообще это была шутка про раковые комментарии в вк, тип они и до пр в гитхабе добрались. Ироничней будь, слыш

Good-Morning commented:
Я отметил основную часть шутки эмоцией, дескать, понял. 
Ну, и на всякий случай неиронично ответил.

mokhov commented:
При поиске не надо сбрасывать состояние приложения. Т.е ввожу я «ка», потом замираю и ввожу «к». Нужно чтобы между «ка» и «как» приложение не сбрасывало состояние в исходное

Commit:
fixed filtering bouncing

Commit:
fixed selection during searching

If a part of the found letters were deleted, searching continues.
Fixed behaviour "selectAll" during searching, mb smth else.
